tests/scalers/aws/aws_cloudwatch_ignore_null_values_false/aws_cloudwatch_ignore_null_values_false_test.go

```
@@ -0,0 +1,196 @@
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  maxReplicaCount: 2
+  minReplicaCount: 0
+  cooldownPeriod: 1
+  triggers:
+    - type: aws-cloudwatch
+      authenticationRef:
+        name: keda-trigger-auth-aws-credentials
+      metadata:
+        awsRegion: {{.AwsRegion}}
+        namespace: {{.CloudWatchMetricNamespace}}
+        dimensionName: {{.CloudWatchMetricDimensionName}}
+        dimensionValue: {{.CloudWatchMetricDimensionValue}}
+        metricName: {{.CloudWatchMetricName}}
+        targetMetricValue: "1"
+        minMetricValue: "1"
+        ignoreNullValues: "false"
+        metricCollectionTime: "120"
+        metricStatPeriod: "60"
+`
+)
+
+var (
+	testNamespace                  = fmt.Sprintf("%s-ns", testName)
+	deploymentName                 = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName               = fmt.Sprintf("%s-so", testName)
+	secretName                     = fmt.Sprintf("%s-secret", testName)
+	cloudwatchMetricName           = fmt.Sprintf("cw-%d", GetRandomNumber())
+	awsAccessKeyID                 = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey             = os.Getenv("TF_AWS_SECRET_KEY")
+	awsRegion                      = os.Getenv("TF_AWS_REGION")
+	cloudwatchMetricNamespace      = "DoesNotExist"
+	cloudwatchMetricDimensionName  = "dimensionName"
+	cloudwatchMetricDimensionValue = "dimensionValue"
+	minReplicaCount                = 0
+)
+
+// This test is to verify that the scaler results in an error state when
+// the metric query returns null values and the ignoreNullValues is set to false.
+func TestCloudWatchScalerWithIgnoreNullValuesFalse(t *testing.T) {
+	ctx := context.Background()
+
+	// setup cloudwatch
+	cloudwatchClient, err := cloudwatch.NewClient(ctx, awsRegion, awsAccessKeyID, awsSecretAccessKey, "")
+	assert.Nil(t, err, "error creating cloudwatch client")
```

```suggestion
	require.Nil(t, err, "error creating cloudwatch client")
```
We try to fail fast when the test requirements have failed